Cache error when using Russian as the system language (and possibly other languages besides English)

Recently, on my Mac (this may be important), I encountered the following header error after a request was cached on disk and the application was restarted:
```java
Caused by: java.lang.IllegalArgumentException: illegal header value: 'чт, 15 сент. 2022 09:21:44 GMT'
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:502)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:486)
	at com.github.mizosoft.methanol.internal.Utils.tryCloneThrowable(Utils.java:169)
	at com.github.mizosoft.methanol.internal.Utils.rethrowExecutionExceptionCause(Utils.java:134)
	at com.github.mizosoft.methanol.internal.Utils.get(Utils.java:220)
	at com.github.mizosoft.methanol.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:162)
	at com.github.mizosoft.methanol.Methanol$InterceptorChain.forward(Methanol.java:1145)
	at com.github.mizosoft.methanol.internal.cache.ChainAdapter.forward(ChainAdapter.java:58)
	at com.github.mizosoft.methanol.internal.cache.RedirectingInterceptor$Exchange.exchange(RedirectingInterceptor.java:118)
	at com.github.mizosoft.methanol.internal.cache.RedirectingInterceptor.exchange(RedirectingInterceptor.java:98)
	... 15 more
Caused by: java.lang.IllegalArgumentException: illegal header value: 'чт, 15 сент. 2022 09:21:44 GMT'
	at com.github.mizosoft.methanol.internal.Validate.requireArgument(Validate.java:47)
	at com.github.mizosoft.methanol.internal.Utils.requireValidHeaderValue(Utils.java:87)
	at com.github.mizosoft.methanol.internal.extensions.HeadersBuilder.lambda$set$7(HeadersBuilder.java:103)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at com.github.mizosoft.methanol.internal.extensions.HeadersBuilder.set(HeadersBuilder.java:103)
	at com.github.mizosoft.methanol.internal.extensions.HeadersBuilder.set(HeadersBuilder.java:96)
	at com.github.mizosoft.methanol.MutableRequest.setHeader(MutableRequest.java:325)
	at com.github.mizosoft.methanol.internal.cache.CacheStrategy.lambda$conditionalize$4(CacheStrategy.java:176)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at com.github.mizosoft.methanol.internal.cache.CacheStrategy.lambda$conditionalize$5(CacheStrategy.java:174)
	at com.github.mizosoft.methanol.MutableRequest.apply(MutableRequest.java:150)
	at com.github.mizosoft.methanol.internal.cache.CacheStrategy.conditionalize(CacheStrategy.java:171)
	at com.github.mizosoft.methanol.internal.cache.CacheInterceptor$Exchange.exchange(CacheInterceptor.java:624)
	at com.github.mizosoft.methanol.internal.cache.CacheInterceptor$Exchange.lambda$exchange$1(CacheInterceptor.java:570)
	at java.base/java.util.Optional.map(Optional.java:260)
	at com.github.mizosoft.methanol.internal.cache.CacheInterceptor$Exchange.lambda$exchange$3(CacheInterceptor.java:568)
	at com.github.mizosoft.methanol.internal.concurrent.CancellationPropagatingFuture.lambda$attachTo$0(CancellationPropagatingFuture.java:100)
	at java.base/java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1187)
	at java.base/java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2341)
	at com.github.mizosoft.methanol.internal.concurrent.CancellationPropagatingFuture.thenCompose(CancellationPropagatingFuture.java:80)
	at com.github.mizosoft.methanol.internal.cache.CacheInterceptor$Exchange.exchange(CacheInterceptor.java:565)
	at com.github.mizosoft.methanol.internal.cache.CacheInterceptor.exchange(CacheInterceptor.java:180)
	... 20 more
```
My cache settings: https://github.com/RedlanceMinecraft/CommonUtils/blob/main/src/main/java/org/redlance/common/utils/requester/Requester.java#L33-L43
The request is sent without errors until caching or if caching is disabled.

--- a/methanol/src/main/java/com/github/mizosoft/methanol/internal/cache/HttpDates.java
+++ b/methanol/src/main/java/com/github/mizosoft/methanol/internal/cache/HttpDates.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024 Moataz Hussein
+ * Copyright (c) 2025 Moataz Hussein
  *
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to deal
@@ -35,6 +35,7 @@ import java.time.format.DateTimeFormatter;
 import java.time.temporal.TemporalAccessor;
 import java.time.temporal.TemporalQueries;
 import java.util.List;
+import java.util.Locale;
 import java.util.Optional;
 
 /** Helpers for parsing/formatting HTTP dates. */
@@ -48,7 +49,12 @@ public class HttpDates {
   private static final List<DateTimeFormatter> FORMATTERS;
 
   static {
-    PREFERRED_FORMATTER = DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss 'GMT'");
+    // Note that we need to create formatters with the US locale explicitly as they'll use the
+    // default locale (e.g. system language) otherwise, which is not what we want (see
+    // https://github.com/mizosoft/methanol/issues/134).
+
+    PREFERRED_FORMATTER =
+        DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss 'GMT'").withLocale(Locale.US);
 
     // These are the formats specified by rfc7231 Section 7.1.1.1 for acceptable HTTP dates.
     FORMATTERS =
@@ -57,14 +63,16 @@ public class HttpDates {
             PREFERRED_FORMATTER,
 
             // Obsolete formats, but allowed by rfc7231.
-            DateTimeFormatter.ofPattern("EEEE, dd-MMM-uu HH:mm:ss 'GMT'"), // rfc850
-            DateTimeFormatter.ofPattern("EEE MMM ppd HH:mm:ss uuuu"), // C's asctime()
+            DateTimeFormatter.ofPattern("EEEE, dd-MMM-uu HH:mm:ss 'GMT'")
+                .withLocale(Locale.US), // rfc850
+            DateTimeFormatter.ofPattern("EEE MMM ppd HH:mm:ss uuuu")
+                .withLocale(Locale.US), // C's asctime()
 
             // A lenient version of the preferred format, with the possibility of zone offsets.
-            DateTimeFormatter.RFC_1123_DATE_TIME,
+            DateTimeFormatter.RFC_1123_DATE_TIME.withLocale(Locale.US),
 
             // The preferred format but with UTC instead of GMT.
-            DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss 'UTC'"));
+            DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss 'UTC'").withLocale(Locale.US));
   }
 
   private HttpDates() {}

--- a/methanol/src/test/java/com/github/mizosoft/methanol/internal/cache/HttpDatesTest.java
+++ b/methanol/src/test/java/com/github/mizosoft/methanol/internal/cache/HttpDatesTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024 Moataz Hussein
+ * Copyright (c) 2025 Moataz Hussein
  *
  * Permission is hereby granted, free of charge, to any person obtaining a copy
  * of this software and associated documentation files (the "Software"), to deal
@@ -33,6 +33,7 @@ import static org.assertj.core.api.InstanceOfAssertFactories.LOCAL_DATE_TIME;
 import com.github.mizosoft.methanol.testing.Logging;
 import java.time.Duration;
 import java.time.ZoneOffset;
+import java.util.Locale;
 import org.junit.jupiter.api.Test;
 
 class HttpDatesTest {
@@ -103,5 +104,25 @@ class HttpDatesTest {
     assertThatObject(tryParseHttpDate("Fri, 09 Sep 2022 23:03:14 GMT").orElse(null))
         .extracting(HttpDates::formatHttpDate)
         .isEqualTo("Fri, 09 Sep 2022 23:03:14 GMT");
+  }
+
+  @Test
+  void defaultLocalDoesNotAffectFormatting() {
+    var currentDefault = Locale.getDefault(Locale.Category.FORMAT);
+    Locale.setDefault(Locale.Category.FORMAT, Locale.FRANCE);
+    try {
+      assertThatObject(tryParseHttpDate("Fri, 09 Sep 2022 23:03:14 GMT").orElse(null))
+          .extracting(HttpDates::formatHttpDate)
+          .isEqualTo("Fri, 09 Sep 2022 23:03:14 GMT");
+
+      assertThat(tryParseHttpDate("Fri, 09 Sep 2022 23:03:14 GMT").orElse(null))
+          .isEqualTo("2022-09-09T23:03:14");
+      assertThat(tryParseHttpDate("Friday, 09-Sep-22 23:03:14 GMT").orElse(null))
+          .isEqualTo("2022-09-09T23:03:14");
+      assertThat(tryParseHttpDate("Fri Sep  9 23:03:14 2022").orElse(null))
+          .isEqualTo("2022-09-09T23:03:14");
+    } finally {
+      Locale.setDefault(Locale.Category.FORMAT, currentDefault);
+    }
   }
 }

